### PR TITLE
Explorer updates

### PR DESCRIPTION
**Pull request by jwildfire**
Expanding the webcodebook.explorer() functionality

Functionality updates (Wiki updates needed): 
- totally re-worked explorer config and output
- added new "files" option to the nav
- added "init" and "complete" callbacks
- added "disabled" option to controlVisibility ("hidden" is now a temporary state that can vary from tab to tab.

### CODE REVIEW

**Comment by jwildfire** [comment]
Updated explorer functionality ready for review.

**Comment by jwildfire** [comment]
Example here: https://github.com/RhoInc/viz-library/pull/44

**Comment by samussiah** [comment]
Controls are occasionally invisible at initialization.

image

**Comment by jwildfire** [comment]
@samussiah Pretty sure I fixed the control visibility on load issue in #153. (So ok to ignore it here)

File: src/explorer/fileListing/onDraw.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      .select('tbody')
+      .selectAll('tr')
+      .filter(f => f.raw == explorer.current)
+      .classed('selected', true);
+
+    //Linkify the labelColumn
+    var labelCells = this.table
+      .selectAll('td')
+      .filter(f => f.col == explorer.config.labelColumn)
+      .classed('link', true)
```
Comment (samussiah) [inline src/explorer/fileListing/onDraw.js]: the styles below should go in the css file

File: src/explorer/setDefaults.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import defaultSettings from './defaultSettings';
+
+export function setDefaults(explorer) {
+  /********************* ignoredColumns *********************/
+  var firstKey = Object.keys(explorer.config.files[0])[0];
```
Comment (samussiah) [inline src/explorer/setDefaults.js]: In my test this removes the label property and thus the label column.

File: src/explorer/makeCodebook.js [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
+  explorer.codebookWrap.selectAll('*').remove();
+
+  //add the Files section to the nav for each config
+  this.current.settings.tabs = this.current.settings.tabs
+    ? d3merge([['files'], this.current.settings.tabs])
+    : ['files', 'codebook', 'listing', 'settings'];
+
+  //set the default tab to the codebook or listing view assuming they are visible
+  if (this.current.event == 'click') {
+    this.current.settings.defaultTab = this.current.settings.tabs.indexOf(
```
Comment (samussiah) [inline src/explorer/makeCodebook.js]: my default tab setting is only effected on the first load; when i choose another file with, say, a default tab of 'listing' i'm taken to the codebook tab

File: src/explorer/makeCodebook.js [hunk 4]
```diff
@@ -1,12 +1,36 @@
...
+  //create the codebook
+  explorer.codebook = webcodebook.createCodebook(
     '.web-codebook-explorer .codebookWrap',
-    meta.settings
+    this.current.settings
   );
-  d3csv(meta.path, function(error, data) {
-    codebook.init(data);
+
+  explorer.codebook.on('complete', function() {
```
Comment (samussiah) [inline src/explorer/makeCodebook.js]: nice callback!  real elevated stuff

File: src/explorer/fileListing/onDraw.js (hunk 1 above)
Comment (jwildfire) [inline src/explorer/fileListing/onDraw.js]: Agreed. See #158.

File: src/explorer/setDefaults.js (hunk 2 above)
Comment (jwildfire) [inline src/explorer/setDefaults.js]: This was a typo - Label column and ignored columns default were transposed. Was fixed in 2nd PR.

File: src/explorer/makeCodebook.js (hunk 3 above)
Comment (jwildfire) [inline src/explorer/makeCodebook.js]: This is intentional, but maybe not ideal. Clicking the file listing automatically loads the codebook - guess we could make that an option at some point, but i think the only reasonable options are the codebook and the data listing.

**Pull request merged**